### PR TITLE
[release-3.0] 3.0.2 documentation updates (#315)

### PR DESCRIPTION
**Pull request by hardys**
This is a backport of https://github.com/suse-edge/suse-edge.github.io/pull/315

* Update configure-network script for Metal3 0.7.3

Switch to the unified config mode, which means mac-address is optional and may be omitted in some environments

* Add 3.0.2 release notes

Co-authored-by: ranjinimn <ranjini.n@suse.com>

* Update versions for 3.0.2

* Add day-2 note for CAPI controller upgrade

3.0.2 updates the RKE2 provider so describe how to upgrade it

* Further version updates

* Bump Rancher images for 2.8.5

* Add kubevirt feature note/link

---------

Co-authored-by: ranjinimn <ranjini.n@suse.com>
(cherry picked from commit b5a0beddf92dd1ec108bbec7b171fc86ddb8b44d)

### CODE REVIEW

File: asciidoc/day2/mgmt-cluster.adoc [hunk 1]
```diff
@@ -542,6 +542,24 @@ metal3-metal3-mariadb-55fd44b648-7fmvk                   1/1     Running   0
...
 # Expected output
-NAME  	NAMESPACE    	REVISION	UPDATED                                	STATUS  	CHART       	APP VERSION
-metal3	metal3-system	2       	2024-06-17 12:43:06.774802846 +0000 UTC	deployed	metal3-0.7.1	1.16.0   
+NAME    NAMESPACE      REVISION  UPDATED                                  STATUS    CHART         APP VERSION
+metal3  metal3-system  2         2024-06-17 12:43:06.774802846 +0000 UTC  deployed  metal3-0.7.3  1.16.0
 ----
+
+== Cluster API upgrade
+
+The Cluster API (CAPI) controllers on a Metal^3^ management cluster are not currently managed via Helm, this section describes the upgrade process.
```
Comment (ranjinimn) [inline asciidoc/day2/mgmt-cluster.adoc:551]: Can you please elaborate more about this section? 
We can mention about not managed via Helm as a note.

File: asciidoc/day2/mgmt-cluster.adoc [hunk 2]
```diff
@@ -542,6 +542,24 @@ metal3-metal3-mariadb-55fd44b648-7fmvk                   1/1     Running   0
...
+
+[NOTE]
+====
+This section assumes you have installed `clusterctl` and configured on your system as described in the <<quickstart-metal3, Metal^3^ quickstart>>
```
Comment (ranjinimn) [inline asciidoc/day2/mgmt-cluster.adoc:555]: This can be a prerequisite.

File: asciidoc/day2/mgmt-cluster.adoc [hunk 3]
```diff
@@ -542,6 +542,24 @@ metal3-metal3-mariadb-55fd44b648-7fmvk                   1/1     Running   0
...
+====
+
+When upgrading to Edge 3.0.2 from any previous version it will be necessary to upgrade the RKE2 providers:
```
Comment (ranjinimn) [inline asciidoc/day2/mgmt-cluster.adoc:558]:
```suggestion
When upgrading to Edge 3.0.2 from any previous version it is necessary to upgrade the RKE2 providers.
```

File: asciidoc/day2/mgmt-cluster.adoc [hunk 4]
```diff
@@ -542,6 +542,24 @@ metal3-metal3-mariadb-55fd44b648-7fmvk                   1/1     Running   0
...
+
+[,bash]
+----
+clusterctl upgrade apply --bootstrap "rke2:v0.4.1" --control-plane "rke2:v0.4.1"
+----
+
+WARNING: Please ensure the versions selected align with those described in the <<id-release-notes, Release Notes>>, usage of other upstream releases is not supported.
```
Comment (ranjinimn) [inline asciidoc/day2/mgmt-cluster.adoc:565]:
```suggestion
WARNING: Ensure the versions selected align with those described in the <<id-release-notes, Release Notes>>, usage of other upstream releases are not supported.
```

File: asciidoc/edge-book/releasenotes.adoc [hunk 5]
```diff
@@ -216,6 +216,102 @@ s| SR-IOV Network Operator s| 1.2.2 s| 1.2.2+up0.1.0 | registry.suse.com/edge/sr
...
+Summary: SUSE Edge 3.0.2 is the second z-stream release in the SUSE Edge 3.0 portfolio.
+
+== New Features
+
+* The Metal^3^ chart now supports static network configuration without any `mac-address`: https://github.com/suse-edge/charts/pull/134[SUSE Edge issue #134]
+* Kubevirt is updated from `1.1.1` to `1.2.2` for details of new features refer to the: https://github.com/kubevirt/kubevirt/releases[Upstream Release Notes]
+
+== Bug & Security Fixes
+
+* The Metal^3^ chart fixes an issue where host reprovisioning may use stale static static network configuration: https://github.com/suse-edge/charts/pull/133[SUSE Edge issue #133]
```
Comment (ranjinimn) [inline asciidoc/edge-book/releasenotes.adoc:232]:
```suggestion
* The Metal^3^ chart fixes an issue where host reprovisioning might use stale static static network configuration: https://github.com/suse-edge/charts/pull/133[SUSE Edge issue #133]
```

File: asciidoc/edge-book/releasenotes.adoc [hunk 6]
```diff
@@ -216,6 +216,102 @@ s| SR-IOV Network Operator s| 1.2.2 s| 1.2.2+up0.1.0 | registry.suse.com/edge/sr
...
+* The RKE2 Cluster API provider fixes an issue when specifying TLS configuration for a local registry: https://github.com/rancher/cluster-api-provider-rke2/pull/357[RKE2 Provider issue #357]
+* The RKE2 Cluster API provider fixes an issue causing rke2-install to error after system reboot: https://github.com/rancher/cluster-api-provider-rke2/pull/346[RKE2 Provider issue #346]
```
Comment (ranjinimn) [inline asciidoc/edge-book/releasenotes.adoc:234]:
```suggestion
* The RKE2 Cluster API provider fixes an issue causing `rke2-install` to error after system reboot: https://github.com/rancher/cluster-api-provider-rke2/pull/346[RKE2 Provider issue #346]
```

File: asciidoc/edge-book/releasenotes.adoc [hunk 7]
```diff
@@ -216,6 +216,102 @@ s| SR-IOV Network Operator s| 1.2.2 s| 1.2.2+up0.1.0 | registry.suse.com/edge/sr
...
+* Kubevirt is updated to include several security fixes: https://www.suse.com/support/update/announcement/2024/suse-su-20242669-1[Kubevirt Update]
+
+== Components Versions
+
+The following table describes the individual components that make up the 3.0.2 release, including the version, the Helm chart version (if applicable), and from where the released artifact can be pulled in the binary format. Please follow the associated documentation for usage and deployment examples.
```
Comment (ranjinimn) [inline asciidoc/edge-book/releasenotes.adoc:239]:
```suggestion
The following table describes the individual components that make up the 3.0.2 release, including the version, the Helm chart version (if applicable), and from where the released artifact can be pulled in the binary format. Follow the associated documentation for usage and deployment examples.
```

File: asciidoc/product/atip-management-cluster.adoc [hunk 8]
```diff
@@ -1437,12 +1437,12 @@ The `clusterctl` and the rest of the files included in the `overrides` folder ca
...
-curl -L https://github.com/rancher-sandbox/cluster-api-provider-rke2/releases/download/v0.2.6/bootstrap-components.yaml
-curl -L https://github.com/rancher-sandbox/cluster-api-provider-rke2/releases/download/v0.2.6/metadata.yaml
+curl -L https://github.com/rancher-sandbox/cluster-api-provider-rke2/releases/download/v0.4.1/bootstrap-components.yaml
+curl -L https://github.com/rancher-sandbox/cluster-api-provider-rke2/releases/download/v0.4.1/metadata.yaml
 
 # control-plane-components (control-plane-rke2)
-curl -L https://github.com/rancher-sandbox/cluster-api-provider-rke2/releases/download/v0.2.6/control-plane-components.yaml
-curl -L https://github.com/rancher-sandbox/cluster-api-provider-rke2/releases/download/v0.2.6/metadata.yaml
+curl -L https://github.com/rancher-sandbox/cluster-api-provider-rke2/releases/download/v0.4.1/control-plane-components.yaml
+curl -L https://github.com/rancher-sandbox/cluster-api-provider-rke2/releases/download/v0.4.1/metadata.yaml
```
Comment (ranjinimn) [inline asciidoc/product/atip-management-cluster.adoc:1445]: Just a question. From 2.6 to 4.1 directly? Are there no other versions in between?

File: asciidoc/quickstart/metal3.adoc [hunk 9]
```diff
@@ -444,7 +445,8 @@ spec:
 # Remaining content as in previous example
 ----
 
-NOTE: although optional in the nmstate API, the mac-address is mandatory for configuration via NM Configurator and must be provided.
+NOTE: In some circumstances the mac-address may be omitted but the `configure-network.sh` script must use the `_all.yaml` filename described above to enable
```
Comment (ranjinimn) [inline asciidoc/quickstart/metal3.adoc:448]:
```suggestion
NOTE: In some circumstances the mac-address can be omitted but the `configure-network.sh` script must use the `_all.yaml` filename described above to enable <<networking-unified, Unified node configuration>> in nm-configurator.
```

Should it be mac-address or `mac-address`?
Should it be nm-configurator. or `nm-configurator.`?

**Review by ranjinimn — COMMENTED** [review]
@hardys I have reviewed the content and provided a few suggestions. Please let me know if you have any questions.

**Comment by hardys** [comment]
@ranjinimn thanks very much for the review - however this is a backport from main of https://github.com/suse-edge/suse-edge.github.io/pull/315 (which you previously approved) so I don't think it's appropriate for me to make any changes in this PR, we'll have to address any concerns via another PR to main first then do another backport.

Are you OK with this iteration landing so we can publish the 3.0.2 release notes today, then we can discuss any follow-up actions?

File: asciidoc/product/atip-management-cluster.adoc (hunk 8 above)
Comment (hardys) [reply asciidoc/product/atip-management-cluster.adoc:1445]: There are versions in between, but the Edge releases don't necessarily consume every version of dependencies like this, we rebase to a newer version to pick up bug fixes (for z-stream updates like this) or features (for y-stream updates like 3.1).

In this case the bugfixes we need landed in 0.4.1 so we rebased to that version and re-tested with the other components of the stack.

**Comment by ranjinimn** [comment]
Sure @hardys You can address it in the next iteration.

**Pull request merged**